Source/santad/EventProviders/SNTDeviceManager.mm

```diff
@@ -0,0 +1,292 @@
+  SNTDeviceManager *app = (__bridge SNTDeviceManager *)context;
+  NSDictionary *props = CFBridgingRelease(DADiskCopyDescription(disk));
+  if (![props[@"DAVolumeMountable"] boolValue]) return;
+
+  [[SNTEventLog logger] logDiskDisappeared:props];
+  if (@available(macos 10.15, *)) [app flushCache];
+}
+
+NSArray<NSString *> *maskToMountArgs(long remountOpts) {
+  NSMutableArray<NSString *> *args = [NSMutableArray array];
+  if (remountOpts & MNT_RDONLY) [args addObject:@"rdonly"];
+  if (remountOpts & MNT_NOEXEC) [args addObject:@"noexec"];
+  if (remountOpts & MNT_NOSUID) [args addObject:@"nosuid"];
+  if (remountOpts & MNT_DONTBROWSE) [args addObject:@"nobrowse"];
+  if (remountOpts & MNT_UNKNOWNPERMISSIONS) [args addObject:@"noowners"];
+  if (remountOpts & MNT_NODEV) [args addObject:@"nodev"];
+  if (remountOpts & MNT_JOURNALED) [args addObject:@"-j"];
+  if (remountOpts & MNT_ASYNC) [args addObject:@"async"];
+  return args;
+}
+
+long mountArgsToMask(NSArray<NSString *> *args) {
+  long flags = 0;
+  for (NSString *arg in args) {
+    if ([arg isEqualToString:@"rdonly"])
+      flags |= MNT_RDONLY;
+    else if ([arg isEqualToString:@"noexec"])
+      flags |= MNT_NOEXEC;
+    else if ([arg isEqualToString:@"nosuid"])
+      flags |= MNT_NOSUID;
+    else if ([arg isEqualToString:@"nobrowse"])
+      flags |= MNT_DONTBROWSE;
+    else if ([arg isEqualToString:@"noowners"])
+      flags |= MNT_UNKNOWNPERMISSIONS;
+    else if ([arg isEqualToString:@"nodev"])
+      flags |= MNT_NODEV;
+    else if ([arg isEqualToString:@"-j"])
+      flags |= MNT_JOURNALED;
+    else if ([arg isEqualToString:@"async"])
+      flags |= MNT_ASYNC;
+  }
+  return flags;
+}
+
+@interface SNTDeviceManager ()
+
+@property DASessionRef diskArbSession;
+@property(readonly, nonatomic) es_client_t *client;
+@property(nonatomic, readonly) dispatch_queue_t esAuthQueue;
+@end
+
+@implementation SNTDeviceManager
+
+- (instancetype)init API_AVAILABLE(macos(10.15)) {
+  self = [super init];
+  if (self) {
+    _blockUSBMount = false;
+
+    dispatch_queue_t disk_queue =
+      dispatch_queue_create("com.google.santad.disk_queue", DISPATCH_QUEUE_SERIAL);
+
+    _esAuthQueue =
+      dispatch_queue_create("com.google.santa.daemon.es_device_auth", DISPATCH_QUEUE_CONCURRENT);
+
+    _diskArbSession = DASessionCreate(NULL);
+    DASessionSetDispatchQueue(_diskArbSession, disk_queue);
+
+    if (@available(macos 10.15, *)) [self initES];
+  }
+  return self;
+}
+
+- (void)initES API_AVAILABLE(macos(10.15)) {
+  while (!self.client) {
+    es_client_t *client = NULL;
+    es_new_client_result_t ret = es_new_client(&client, ^(es_client_t *c, const es_message_t *m) {
+      // Set timeout to 5 seconds before the ES deadline.
+      [self handleESMessageWithTimeout:m
+                            withClient:c
+                               timeout:dispatch_time(m->deadline, NSEC_PER_SEC * -5)];
+    });
+    [self flushCache];
```

Comment:
Can remove this call. ES automatically clears caches when new clients connect (or existing clients disconnect)

Reply:
Done.